Team_Roster/immaculateGridCalculations/gridSolver.py

```diff
@@ -278,6 +276,130 @@ def getOneTeamPlayers():
         .having(func.count(func.distinct(Batting.teamID)) == 1)
     )
 
+#------------------------------------------------------------------------------
+# All players with a season ERA less than a certain amount while on a certain team
+#Parameters
+# - maxERA (int): The maximum season ERA required
+# Notes:
+def getPlayerSeasonERA(maxERA):
+    return (
+        db.session.query(
+            Pitching.playerID.label("playerID"),
+            Pitching.teamID.label("teamID")
+        )
+        .group_by(
+            Pitching.playerID,
+            Pitching.yearID
+        )
+        .having(
+            (
+                func.sum(Pitching.p_ER) /
+                (func.sum(Pitching.p_IPOuts)/3) * 9
+            )
+        ) <= maxERA
+    )
+
+# All players with a season run over a certain amount while one a certain team
+# Parameters:
+# - minRun (int): The minimum season run required
+# Notes:
+def getPlayerSeasonRun(minRun):
+    return (
+        db.session.query(
+            Batting.playerID.label("playerID"),
+            Batting.teamID.label("teamID")
+        )
+        .group_by(
+            Batting.playerID,
+            Batting.yearID
+        )
+        .having(
+            Batting.b_R >= minRun
+        )
+    )
+
+# All players with a career save over a certain amount
+# Parameters:
+# -svNum (int) The minimum SV required
+# Notes:
+def getPlayerSVByCareer(svNum):
+    return (
+        db.session.query(
+            Pitching.playerID.label("playerID"),
+            Pitching.teamID.label("teamID")
+        )
+        .group_by(
+            Pitching.playerID,
+            Pitching.teamID
+        )
+        .having(
+            func.sum(Pitching.p_SV) >= svNum
+        )
+    )
+
+# All players with a season double over a certain amount
+# Parameters:
+# - dblNum (int) The minimum double required
+# Notes
+def getPlayerDBLBySeason(dblNum):
+    return (
+        db.session.query(
+            Batting.playerID.label("playerID"),
+            Batting.teamID.label("teamID")
+        )
+        .filter(
+            Batting.b_2B >= dblNum
+        )
+        .group_by(
+            Batting.playerID,
+            Batting.teamID,
+            Batting.yearID
+        )
+    )
+
+# All players in the hall of fame
+# Parameters:
+# N/A
+# Notes:
+def getPlayerHallOfFame():
+    return (
+        db.session.query(
+            HallOfFame.playerID.label("playerID"),
+            Batting.teamID.label("teamID")
+        )
+        .join(
+            Batting,
+            HallOfFame.playerID == Batting.playerID
+        )
+        .filter(
+            HallOfFame.inducted == 'y'
+        )
+        .group_by(
+            HallOfFame.playerID,
+            Batting.teamID
+        )
+    )
+
+# All players that threw a no-hitter
+# Parameters:
+# N/A
+# Notes:
+def getPlayerNoHitter():
```

Comment:
I don't think this is a correct way of checking this, this would get a pitcher who has 0 hits and 27 outs _in an entire stint_...
We may not have a way to calculate this...